### ISSUE TITLE
PackageImports extension doesn't work....

### ISSUE THREAD

**Issue by alcidesv** [gibiansky/IHaskell#587]
At present I can not disambiguate between packages for importing certain modules:

```
-- This doesn't work:   ":ext PackageImports", the kernel hangs
-- This doesn't work:   "{-# LANGUAGE PackageImports #-}", the kernel hangs
import qualified "crypto-api" Crypto.Random                                         as CR
```

I ended up using the following hack:

```
 :!rm himports.hs
 :!echo "{-# LANGUAGE PackageImports #-}" >> himports.hs
 :!echo "module MouseBoxImports( module Crypto.Random ) where" >> himports.hs
 :!echo "import  \"crypto-api\" Crypto.Random" >> himports.hs
 :load himports
```
##

**Comment by sumitsahrawat** [comment]
I am unable to recreate this. Which version, install method etc. are you using?

**Comment by alcidesv** [comment]
I'm using the stack installation method.

**Comment by sumitsahrawat** [comment]
There have been some updates relating to how stack is handled. Are you using the latest ihaskell from github master, or the one using `stack install ihaskell` (which I think installs ihaskell-0.6.5.0)?

Using the latest IHaskell from github master, here is an example session:

![image](https://cloud.githubusercontent.com/assets/5645741/10411692/ca22c318-6f8b-11e5-8f30-9339e28318da.png)

Installed using `stack install ihaskell` in the repository, and executed as `jupyter notebook`.

**Comment by alcidesv** [comment]
Ok, I installed per your recommendation the last version straight from a repository clone. I can confirm that 
   
     import "base" Prelude 

works well,

but 

     :ext PackageImports
    import  "crypto-api" Crypto.Random

still kills the kernel. Here is what comes in the log:

    [I 17:13:42.122 NotebookApp] Saving file at /testyard/RSA.ipynb
    ihaskell: Ambiguous interface for ‘Crypto.Random’: it was found in multiple packages: crypto-api-0.13.2 cryptonite-0.7

    [I 17:14:53.919 NotebookApp] KernelRestarter: restarting kernel (1/5)
    WARNING:root:kernel 4d782d88-1fed-49c1-a847-771e6cff8da5 restarted

**Comment by vaibhavsagar** [comment]
This was fixed in GHC 8.4, see https://ghc.haskell.org/trac/ghc/ticket/13168. I tried your example and it worked in IHaskell.
